### PR TITLE
Realted post field added

### PR DESCRIPTION
**Pull request by dipen30**

### CODE REVIEW

**Comment by yakky** [comment]
I am +1 on adding the ``related`` field
I am not sure about adding such a strong dependency as django-admin-sortable, also because there are many sorting / ordering applications and tying djangocms-blog to a specific implementation can hamper its inclusion in projects using other sorting implementations
Also, I don't like the "order" field as the order should be on the related through model itself.
Maybe reworking this using m2m-sortable could be a better approach

**Comment by yakky** [comment]
@dipen30 nice addition. do you have any opinion on my comments regarding the sorting library?

**Comment by dipen30** [comment]
@yakky You are right We can either user  django-sortedm2m.

But I think we can create one abstract class and give user a little flexibility to add few more fields based on need Like you have done in Plugin Models

https://github.com/nephila/djangocms-blog/blob/develop/djangocms_blog/models.py#L394

**Comment by yakky** [comment]
@dipen30 wow, you are going the bold way! 
Making post abstract might be complex: swappable models are not easy to implement right and usually require a lot of work, even if the reduced blog surface should not make it very hard to do that.
I'd like to handle the two cases separately, though: related posts definitely is a core feature for a blog.
Abstract base model can then be implemented separately

**Comment by dipen30** [comment]
@yakky I have removed ordering related code and will make another pull request for **django-sortedm2m** if required

**Comment by yakky** [comment]
@dipen30 I welcome relatedm2m in this very PR. Just abstract models looked a bit off in this PR. Thanks for your work and ideas!

File: djangocms_blog/models.py [hunk 1]
```diff
@@ -171,7 +172,8 @@ class Post(KnockerModel, ModelMeta, TranslatableModel):
 
     translations = TranslatedFields(
         title=models.CharField(_('title'), max_length=255),
-        slug=models.SlugField(_('slug'), max_length=255, blank=True, db_index=True),
+        slug=models.SlugField(_('slug'), max_length=255, blank=True,
+                              db_index=True, allow_unicode=True),
```
Comment (yakky) [inline djangocms_blog/models.py]: This is not supported in Django 1.8, unfortunately
Do we have an alternative approach?

File: djangocms_blog/models.py [hunk 2]
```diff
@@ -171,7 +172,8 @@ class Post(KnockerModel, ModelMeta, TranslatableModel):
 
     translations = TranslatedFields(
         title=models.CharField(_('title'), max_length=255),
-        slug=models.SlugField(_('slug'), max_length=255, blank=True, db_index=True),
+        slug=models.SlugField(_('slug'), max_length=255, blank=True,
```
Comment (yakky) [inline djangocms_blog/models.py]: This does dot work in Django 1.8 unfortunately. Do we have an alternative to support the option in 1.9+

File: djangocms_blog/admin.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ class PostAdmin(PlaceholderAdminMixin, FrontendEditableAdminMixin,
     enhance_exclude = ('main_image', 'tags')
     _fieldsets = [
         (None, {
-            'fields': [['title', 'categories', 'publish', 'app_config']]
+            'fields': [['title', 'categories', 'publish', 'app_config'], ['related', ]]
```
Comment (yakky) [inline djangocms_blog/admin.py]: I would move ``related`` this into a dedicated fieldset. The list can grow quite long and can be inconvenient having it just above the abstract

**Comment by yakky** [comment]
@dipen30 I like this! :) check the comments above. Only minor things, actually.

File: djangocms_blog/admin.py (hunk 3 above)
Comment (yakky) [inline djangocms_blog/admin.py]: Thanks for addressing this!

File: djangocms_blog/models.py (hunk 2 above)
Comment (dipen30) [inline djangocms_blog/models.py]: To make it compatible for 1.8+ we can use following library 
https://github.com/neithere/django-autoslug

What do you think ?

Comment (yakky) [inline djangocms_blog/models.py]: A bit wary about adding another dependency for such a small feature. Can we just pass the argument only in 1.9 and above?

**Comment by yakky** [comment]
@dipen30 Nice, thanks for the latest changes!
I'm complete the review and merge this shortly

**Comment by kynsi** [comment]
Is it possible to add a feature to turn this on/off? Or maybe limit the number of posts being displayed? With a lot of posts (>100) adding new posts becomes a bit messy..

**Comment by yakky** [comment]
@kynsi It totally makes sense. You can test the linked PR
In adds a `BLOG_USE_RELATED` setting and the linked per-apphook setting to enable / disable related posts

**Comment by kynsi** [comment]
How do I use the related articles in a template? 

`{% for related in post.related %}`

doesn't work

**Comment by kynsi** [comment]
Never mind.. This works:

`{% for related in post.related.all %}`

**Pull request merged**